Makefile

```diff
@@ -47,135 +35,187 @@ else
+$(OUT_BPF): $(DOCKER_BUILDER) | $(OUT_DIR)
+	$(call docker_builder_make,$@)
+endif
+
+# libbpf build:
+.PHONY: libbpf
+libbpf: $(LIBBPF_HEADERS) $(LIBBPF_OBJ)
 
-libbpf_compile_tools = $(CMD_LLC) $(CMD_CLANG)
+check_%:
+	@command -v $* >/dev/null || (echo "missing required tool $*" ; false)
+
+libbpf_compile_tools = $(CMD_LLC) $(CMD_CC)
 .PHONY: libbpf_compile_tools
 $(libbpf_compile_tools): % : check_%
 
 $(LIBBPF_SRC):
 	test -d $(LIBBPF_SRC) || (echo "missing libbpf source - maybe do 'git submodule init && git submodule update'" ; false)
 
 $(LIBBPF_HEADERS) $(LIBBPF_HEADERS)/bpf $(LIBBPF_HEADERS)/linux: | $(OUT_DIR) libbpf_compile_tools $(LIBBPF_SRC)
-	$(MAKE) -C $(LIBBPF_SRC) install_headers install_uapi_headers DESTDIR=$(abspath $(OUT_DIR))/libbpf
+	$(MAKE) -C $(LIBBPF_SRC) CC="$(CMD_CC)" CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)" install_headers install_uapi_headers DESTDIR=$(abspath $(OUT_DIR))/libbpf/$(ARCH)
 
 $(LIBBPF_OBJ): | $(OUT_DIR) libbpf_compile_tools $(LIBBPF_SRC)
-	$(MAKE) -C $(LIBBPF_SRC) OBJDIR=$(abspath $(OUT_DIR))/libbpf BUILD_STATIC_ONLY=1
+	$(MAKE) -C $(LIBBPF_SRC) CC="$(CMD_CC)" CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)" OBJDIR=$(abspath $(OUT_DIR))/libbpf/$(ARCH) BUILD_STATIC_ONLY=1
 
 $(VMLINUX):
 	bpftool btf dump file /sys/kernel/btf/vmlinux format c > $@
 
-.PHONY: libbpf
-libbpf: $(LIBBPF_HEADERS) $(LIBBPF_OBJ)
+# static analysis:
+lint: check-license go/lint bpf/lint
 
-.PHONY: bpf
-bpf: $(OUT_BPF)
+.PHONY: check-license
+check-license:
+	./scripts/check-license.sh
 
-ifndef DOCKER
-$(OUT_BPF): $(BPF_SRC) | $(OUT_DIR)
-	mkdir -p $(OUT_BPF_DIR)
-	$(MAKE) -C bpf build
-	cp bpf/target/bpfel-unknown-none/release/cpu-profiler $(OUT_BPF)
-else
-$(OUT_BPF): $(DOCKER_BUILDER) | $(OUT_DIR)
-	$(call docker_builder_make,$@)
-endif
+.PHONY: go/lint
+go/lint:
+	$(GO_ENV) golangci-lint run
+
+.PHONY: bpf/lint
+bpf/lint:
+	$(MAKE) -C bpf lint
 
 test/profiler: $(GO_SRC) $(LIBBPF_HEADERS) $(LIBBPF_OBJ) bpf
-	sudo $(go_env) go test $(SANITIZERS) -v $(shell go list ./... | grep "pkg/profiler")
+	sudo $(GO_ENV) $(CGO_ENV) $(GO) test $(SANITIZERS) -v $(shell $(GO) list ./... | grep "pkg/profiler")
 
 .PHONY: test
 ifndef DOCKER
-test: $(GO_SRC) $(LIBBPF_HEADERS) $(LIBBPF_OBJ) build test/profiler
-	$(go_env) go test $(SANITIZERS) -v $(shell go list ./... | grep -v "internal/pprof" | grep -v "pkg/profiler" | grep -v "e2e")
+test: $(GO_SRC) $(LIBBPF_HEADERS) $(LIBBPF_OBJ) $(OUT_BPF) build test/profiler
+	$(GO_ENV) $(CGO_ENV) $(GO) test $(SANITIZERS) -v $(shell $(GO) list ./... | grep -v "internal/pprof" | grep -v "pkg/profiler" | grep -v "e2e")
 else
 test: $(DOCKER_BUILDER)
 	$(call docker_builder_make,$@)
 endif
 
+.PHONY: format
+format: go/fmt bpf/fmt
+
+.PHONY: bpf/fmt
+bpf/fmt:
+	$(MAKE) -C bpf format
+
+.PHONY: go/fmt
+go/fmt:
+	gofumpt -l -w $(shell $(GO) list ./... | grep -E -v "internal/pprof|internal/go" | sed 's#^github.com/parca-dev/parca-agent/##')
+	$(GO) fmt $(shell $(GO) list ./... | grep -E -v "internal/pprof|internal/go")
```

Comment:
`golangci-lint run --fix` can do that, it avoids requiring a standalone `gofumpt` binary, should we have a `go/lint-fix` target instead?